interfaces/cython/cantera/test/test_thermo.py

```diff
@@ -1803,3 +1803,13 @@ def test_slice_SolutionArray(self):
         soln = ct.SolutionArray(self.gas, 10)
         arr = soln[2:9:3]
         self.assertEqual(len(arr.T), 3)
+
+        #tests for edge cases in sliced solution array
+        arr1 = soln[1:1]
+        self.assertEqual(arr1.n_elements, 3)
+
+        arr2 = soln[slice(0)]
+        self.assertEqual(arr2.n_species, 9)
+
+        arr3 = soln[None:0]
+        self.assertEqual(arr3.n_reactions, 28)
```

Comment:
I think the tests need to cover properties other than scalars that are independent of the slice. For instance, shouldn't all of these cases generate zero-length slices? I get some worrying output from taking such slices:
```python
>>> gas = ct.Solution('h2o2.yaml')
>>> s = ct.SolutionArray(gas, 5)
>>> s.TP = [400,500,600,700,800], ct.one_atm
>>> s[:0].T
array(101325.)
>>> s[1:1].P
array(101325.)
>>> s[3:3].X
array([0.00e+000, 0.00e+000, 0.00e+000, 0.00e+000, 0.00e+000, 9.88e-321,
       0.00e+000, 0.00e+000, 4.94e-324])
```

Reply:
@speth I guess the problem was with the default value of states. I guess the update should work.